File: time-conversion/time-conversion.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func timeConversion(time string) string {
	split := strings.Split(time, ":")
	hhUnformatted := split[0][0:2]
	mm := split[1]
	ssUnformatted := split[2]

	hh := hhUnformatted
	if strings.Contains(ssUnformatted, "PM") {
		hhUnformattedN, err := strconv.Atoi(hhUnformatted)
		if err != nil {
			return ""
		}
		hh = strconv.Itoa(hhUnformattedN + 12)
	}
	if hh == "24" {
		hh = "12"
	} else if hh == "12" {
		hh = "00"
	}

	ss := ssUnformatted[0:2]

	return fmt.Sprintf(
		"%s:%s:%s",
		hh,
		mm,
		ss,
	)
}

File: time-conversion/TimeConversion.java
public class TimeConversion {

  public static String timeConversion(String time) {
    String[] split = time.split(":");
    String hhUnformatted = split[0].substring(0, 2);
    String mm = split[1];
    String ssUnformatted = split[2];

    String hh = hhUnformatted;
    if (ssUnformatted.contains("PM")) {
      int hhUnformattedN = Integer.parseInt(hhUnformatted);
      hh = String.valueOf(
          hhUnformattedN + 12);
    }
    if (hh.equals("24")) {
      hh = "12";
    } else if (hh.equals("12")) {
      hh = "00";
    }

    String ss = ssUnformatted.substring(0, 2);

    return String.format(
     "%s:%s:%s",
      hh,
      mm,
      ss
    );
  }

}
